Formula/idris2.rb

```diff
@@ -1,31 +1,98 @@
+      patch :DATA
+    end
+
+    # Fix detection of Apple Silicon. Remove in the next release
+    patch do
+      on_arm do
+        url "https://github.com/idris-lang/Idris2/commit/21ca9066f13e3b9f8c6be28774a91f43bb68d51d.patch?full_index=1"
+        sha256 "468b0afc6f81b7709ba1cfd63d49e9183b5b3ba718e0af774405d3b750f2616d"
+      end
+    end
+  end
+
   bottle do
     sha256 cellar: :any,                 big_sur:      "39fe90502cd4a8a064b4f69bcc5b4e99295826cad363092b47d3353ed41db2da"
     sha256 cellar: :any,                 catalina:     "f29bcc24fc1a0581eb3f17823641c5444acc4df10e93c17eae573deab9a18fa7"
     sha256 cellar: :any_skip_relocation, x86_64_linux: "fdd19a501c9b80da87fd4519ed43e9ed31647e17e03d582392d2b3d1517f7c74"
   end
 
-  depends_on "coreutils" => :build
   depends_on "gmp" => :build
-  depends_on "chezscheme"
-  uses_from_macos "zsh" => :build, since: :mojave
+
+  on_high_sierra :or_older do
+    depends_on "zsh" => :build
+  end
+
+  # Use Racket fork of Chez Scheme for Apple Silicon support while main formula lacks support.
+  # https://github.com/idris-lang/Idris2/blob/main/INSTALL.md#installing-chez-scheme-on-apple-silicon
+  on_arm do
+    depends_on "lz4"
+
+    on_linux do
+      depends_on "ncurses"
+      depends_on "zlib"
+    end
+
+    resource "chezscheme" do
+      url "https://github.com/racket/ChezScheme.git",
+          tag:      "racket-v8.6",
+          revision: "9383dda64db9f430a95bb5cf014af2afdc71fb0c"
+    end
+  end
+
+  on_intel do
+    depends_on "chezscheme"
+  end
 
   def install
+    scheme = if Hardware::CPU.arm?
+      resource("chezscheme").stage do
+        rm_r %w[lz4 zlib]
+        args = %w[LZ4=-llz4 ZLIB=-lz]
+
+        system "./configure", "--pb", *args
+        system "make", "auto.bootquick"
+        system "./configure", "--disable-x11",
+                              "--installprefix=#{libexec}/chezscheme",
+                              "--installschemename=chez",
+                              "--threads",
+                              *args
+        system "make"
+        system "make", "install"
+      end
+      libexec/"chezscheme/bin/chez"
+    else
+      Formula["chezscheme"].opt_bin/"chez"
+    end
+
     ENV.deparallelize
-    scheme = Formula["chezscheme"].bin/"chez"
+    ENV["CHEZ"] = scheme
     system "make", "bootstrap", "SCHEME=#{scheme}", "PREFIX=#{libexec}"
     system "make", "install", "PREFIX=#{libexec}"
-    bin.install_symlink libexec/"bin/idris2"
-    lib.install_symlink Dir["#{libexec}/lib/#{shared_library("*")}"]
-    generate_completions_from_executable(bin/"idris2", "--bash-completion-script", "idris2",
+    if Hardware::CPU.arm?
+      (bin/"idris2").write_env_script libexec/"bin/idris2", CHEZ: "${CHEZ-#{scheme}}"
+    else
+      bin.install_symlink libexec/"bin/idris2"
+    end
+    lib.install_symlink Dir[libexec/"lib"/shared_library("*")]
+    generate_completions_from_executable(libexec/"bin/idris2", "--bash-completion-script", "idris2",
```

Comment:
Env scripts don't seem to work inside install (maybe exec permissions?), so just directly running.